crates/bevy_ecs/src/query/default_filters.rs

```diff
@@ -0,0 +1,167 @@
+use crate as bevy_ecs;
+use crate::{component::ComponentId, query::FilteredAccess};
+use bevy_ecs_macros::Resource;
+
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+enum FilterKind {
+    With,
+    Without,
+}
+
+#[derive(PartialEq, Eq, Debug)]
+struct DefaultFilter {
+    component_id: ComponentId,
+    kind: FilterKind,
+}
+
+/// A list of default query filters, these can be used to globally exclude entities from queries.
+/// Each individual filter is only applied to queries that don't mention that component.
+///
+/// These filters are only applied to queries initialized after updating this resource,
+/// it should most likely only be modified before the app starts.
+#[derive(Resource, Default)]
+pub struct DefaultQueryFilters(Vec<DefaultFilter>);
```

Comment:
Maybe insert it by default so plugin authors aren't needing to check `Option`s all the time to avoid overwriting their default filters.

Reply:
Where do we want to insert it? Looks like we have nothing fitting in `bevy_core` ... But we could add a new plugin there ... Or it could be added to `App:default()`

Reply:
App works since it's just this atm unless someone else knows somewhere better.

Reply:
Might want to move it into its own plugin if we decide to add a bunch of default filters (like excluding window & systems).